Tella/Data/Networking/Repositories/UwaziServerRepositories.swift

```diff
@@ -78,6 +124,136 @@ struct UwaziServerRepository: WebRepository {
+                getSetting,
+                getDictionary,
+                getTranslation
+            )
+            .tryMap({ (templateResult, settings, dictionary, translationResult) in
+                let templates = templateResult.rows
+                let translations = translationResult.rows
+                let dictionary = dictionary.rows
+                let settings: UwaziSetting = settings
+
+                // Maps the options to the property of the template
+                self.handleMapping(templates, dictionary)
+                // Check whether the server instance is public and if public then only use the whitelisted templates are added to resultTemplates
+                let resultTemplates = self.getAllowedTemplates(server: server, settings: settings, templates: templates)
+                self.translate(locale: locale, resultTemplates: resultTemplates, translations: translations)
+                let originalTemplate = resultTemplates.map { template in
+                    return CollectedTemplate(serverId: serverID,
+                                             templateId: template.id,
+                                             serverName: server.name ?? "",
+                                             username: server.username,
+                                             entityRow: template,
+                                             isDownloaded: false,
+                                             isFavorite: false,
+                                             isUpdated: false)
+                }
+                return originalTemplate
+            })
+            .eraseToAnyPublisher()
+        }
+        // Handle error case
+        return Fail(error: APIError.unexpectedResponse).eraseToAnyPublisher()
+    }
+
+    fileprivate func getAllowedTemplates(server: Server, settings: UwaziSetting, templates: [UwaziTemplateRow]?) -> [UwaziTemplateRow] {
+        // Check whether the server instance is public and if public then only use the whitelisted templates are added to resultTemplates
+        var tempTemplates: [UwaziTemplateRow] = []
+        if !self.isPublic(server: server) {
+            tempTemplates = templates ?? []
+        } else {
+            if !settings.allowedPublicTemplates.isEmpty {
+                templates?.forEach { row in
+                    settings.allowedPublicTemplates.forEach { id in
+                        if row.id == id {
+                            tempTemplates.append(row)
+                        }
+                    }
+                }
+            }
+        }
+        return tempTemplates
+    }
+
+    fileprivate func isPublic(server: Server) -> Bool {
+        return (server.username?.isEmpty ?? true) || (server.password?.isEmpty ?? true)
+    }
+
+    // Maps the options to the property of the template
+    fileprivate func handleMapping(_ templates: [UwaziTemplateRow]?, _ dictionary: [UwaziDictionaryRow]?) {
+        templates?.forEach { template in
```

Comment:
For this we are just taking the value from the` dictionaryItem.values` and assigning to the `template.values` so compact is not necessary has it will return something after mapping and we do not need anything returned so. Both the compactMap and ForEach has same time complexity in this case i.e O(n) so there will not also be any performance bottleneck.